code/modules/reagents/reagent_containers/food/cans.dm

```diff
@@ -45,8 +61,39 @@
 		fuseoverlay.pixel_x = can_size_overrides["x"]
 		fuseoverlay.pixel_y = can_size_overrides["y"]
 		add_overlay(fuseoverlay)
+	if(bombcasing)
+		var/image/casingoverlay = image('icons/obj/fuses.dmi', icon_state = "pipe_bomb")
+		add_overlay(casingoverlay)
 
 /obj/item/reagent_containers/food/drinks/cans/attackby(obj/item/W, mob/user)
+	if(istype(W, /obj/item/grenade/chem_grenade/large))
+		var/obj/item/grenade/chem_grenade/grenade_casing = W
+		if(!grenade_casing.detonator && !length(grenade_casing.beakers))
+			bombcasing = BOMBCASING_LOOSE
+			desc = "A grenade casing with \a [name] slotted into it."
+			if(fuselength)
+				desc += " It has some steel wool stuffed into the opening."
+			user.visible_message(SPAN_NOTICE("[user] slots \the [grenade_casing] over \the [name]."), SPAN_NOTICE("You slot \the [grenade_casing] over \the [name]."))
+			desc = "A grenade casing with \a [name] slotted into it."
+			qdel(grenade_casing)
+			update_icon()
+
+	if(W.isscrewdriver() && bombcasing)
+		if(bombcasing == BOMBCASING_LOOSE)
+			bombcasing = BOMBCASING_SECURE
+			shrapnelcount = 14
+			user.visible_message(SPAN_NOTICE("[user] tightens the grenade casing around \the [name]."), SPAN_NOTICE("You tighten the grenade casing around \the [name]."))
+		else if(bombcasing == BOMBCASING_SECURE)
+			bombcasing = 0
```

Comment:
Usually when we have defines like you've done with `BOMBCASING_LOOSE` `BOMBCASING_SECURE` we also define 0, maybe something like `BOMBCASING_EMPTY`.

Then you can make the `if` statement on 81 (and the other `if(bombcasing)` checks) something like:
`if(W.isscrewdriver() && bombcasing > BOMBCASING_EMPTY`

and this line 87
`bombcasing = BOMBCASING_EMPTY`